lib/barnes/instruments/puma_backlog.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Barnes
+  module Instruments
+    class PumaBacklog
+      def initialize(sample_rate=nil)
+      end
+
+      def valid?
+        defined?(Puma) &&
+          Puma.respond_to?(:stats) &&
+          ENV["DYNO"].start_with?("web")
```

Comment:
This line is causing problems for local development (where `ENV["DYNO"]` would be `nil`). Should this have a `ENV["DYNO"] && ` check here?

We ran into this problem when trying to follow the setup instructions on Code Triage, which is using this branch:

```
heroku local -f Procfile.development
[OKAY] Loaded ENV .env File as KEY=VALUE Format
13:51:47 web.1      |  [29170] Puma starting in cluster mode...
13:51:47 web.1      |  [29170] * Version 3.11.3 (ruby 2.5.0-p0), codename: Love Song
13:51:47 web.1      |  [29170] * Min threads: 5, max threads: 5
13:51:47 web.1      |  [29170] * Environment: development
13:51:47 web.1      |  [29170] * Process workers: 2
13:51:47 web.1      |  [29170] * Preloading application
13:51:49 web.1      |  [29170] ! Unable to load application: NoMethodError: undefined method `start_with?' for nil:NilClass
13:51:49 web.1      |  /Users/engineering/.rvm/gems/ruby-2.5.0/bundler/gems/barnes-c38e6b0beba7/lib/barnes/instruments/puma_backlog.rb:12:in `valid?': undefined method `start_with?' for nil:NilClass (NoMethodError)
...
```

Reply:
Great call, i'll update. Thanks for the note, also thanks for the CS content you put out!

Reply:
Yay! Thanks for the quick reply and the kind words. 🙂